cl/phase1/core/state/accessors.go

```diff
@@ -277,22 +338,25 @@ func ExpectedWithdrawals(b abstract.BeaconState, currentEpoch uint64) []*cltypes
 			}
 			withdrawals = append(withdrawals, newWithdrawal)
 			nextWithdrawalIndex++
-		} else if isPartiallyWithdrawableValidator(b.BeaconConfig(), currentValidator, currentBalance) { // Check if the validator is partially withdrawable
+		} else if isPartiallyWithdrawableValidator(b, currentValidator, currentBalance) { // Check if the validator is partially withdrawable
 			// Add a new withdrawal with the validator's withdrawal credentials and balance minus the maximum effective balance
+			amount := currentBalance - b.BeaconConfig().MaxEffectiveBalance
```

Comment:
can't this potentially underflow?

Reply:
in `isPartiallyWithdrawableValidator` https://github.com/erigontech/erigon/pull/12418/files#diff-e354369205a9046bc4ccd2d5e1630441b624dfb4f4baf3ca91fb055c908e5010R119
it's guaranteed `currentBalance` must be greater than `b.BeaconConfig().MaxEffectiveBalance`